python/lsst/cell_coadds/_fits.py

```diff
@@ -0,0 +1,269 @@
+    cell_id = fits.Column(name="cell_id", format="2I", array=array)
+
+    array = np.array([cell.outer.image.array for cell in self.cells.values()])
+    unit_array = np.array([cell.common.units.name for cell in self.cells.values()])
+    image = fits.Column(
+        name="image",
+        unit=unit_array[0],
+        format=f"{np.product(array[0].shape)}D",
+        dim=str(array[0].shape),
+        array=array,
+    )  # make it a dask array for performance?
+
+    array = np.array([cell.outer.mask.array for cell in self.cells.values()])
+    mask = fits.Column(
+        name="mask", format=f"{np.product(array[0].shape)}I", dim=str(array[0].shape), array=array
+    )
+
+    array = np.array([cell.outer.variance.array for cell in self.cells.values()])
+    variance = fits.Column(
+        name="variance", format=f"{np.product(array[0].shape)}E", dim=str(array[0].shape), array=array
+    )
+
+    array = [cell.psf_image.array for cell in self.cells.values()]
+    psf = fits.Column(
+        name="psf", format=f"{np.product(array[0].shape)}D", dim=str(array[0].shape), array=array
+    )
+
+    col_defs = fits.ColDefs([cell_id, image, mask, variance, psf])
+    hdu = fits.BinTableHDU.from_columns(col_defs)
+
+    grid_cell_size = self.grid.cell_size
+    grid_shape = self.grid.shape
+    grid_min = self.grid.bbox.getMin()
+    grid_cards = {
+        "GRCELL1": grid_cell_size.x,
+        "GRCELL2": grid_cell_size.y,
+        "GRSHAPE1": grid_shape.x,
+        "GRSHAPE2": grid_shape.y,
+        "GRMIN1": grid_min.x,
+        "GRMIN2": grid_min.y,
+    }
+    hdu.header.extend(grid_cards)
+
+    outer_cell_size_cards = {"OCELL1": self.outer_cell_size.x, "OCELL2": self.outer_cell_size.y}
+    hdu.header.extend(outer_cell_size_cards)
+
+    psf_image_size_cards = {"PSFSIZE1": self.psf_image_size.x, "PSFSIZE2": self.psf_image_size.y}
+    hdu.header.extend(psf_image_size_cards)
+
+    inner_bbox_cards = {
+        "INBBOX11": self.inner_bbox.minX,
+        "INBBOX12": self.inner_bbox.minY,
+        "INBBOX21": self.inner_bbox.maxX,
+        "INBBOX22": self.inner_bbox.maxY,
+    }
+    hdu.header.extend(inner_bbox_cards)
+
+    wcs = self.common.wcs
+    wcs_cards = wcs.getFitsMetadata().toDict()
+    primary_hdu = fits.PrimaryHDU()
+    primary_hdu.header.extend(wcs_cards)
+
+    hdu.header["TUNIT1"] = self.common.units.name
+    hdu.header["BAND"] = self.common.band  # assumed to be the same as self.common.identifers.band
+    hdu.header["SKYMAP"] = self.common.identifiers.skymap.name  # accessing name should not be required
+    hdu.header["TRACT"] = self.common.identifiers.tract
+    hdu.header["PATCH_X"] = self.common.identifiers.patch.x
+    hdu.header["PATCH_Y"] = self.common.identifiers.patch.y
+
+    hdu_list = fits.HDUList([primary_hdu, hdu])
+    hdu_list.writeto("test.fits")
+
+
+# Attach the writeFits function to the MultipleCellCoadd as a routine.
+MultipleCellCoadd.writeFits = writeFits
```

Comment:
I'd like the idea of the writing and reading code being colocated in the same file, since changes to one method can easily be propagated to the other. So I'm adding a method of `MultipleCellCoadd` here as opposed to in `_multiple_cell_coadd.py`.